content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,677 @@
+  where: { name: 'Alice' },
+}).catch(adaptError(() => new MyCustomError())
+```
+
+#### Replacing <inlinecode>rejectOnNotFound</inlinecode> enabled at the client level.
+
+For cases like these below:
+
+```js
+// Example 1
+const prisma = new PrismaClient({
+  rejectOnNotFound: true,
+})
+
+// Example 2
+const prisma = new PrismaClient({
+  rejectOnNotFound: {
+    findUnique: true,
+  },
+})
+```
+
+You will need to update your codebase to use `findUniqueOrThrow` and `findFirstOrThrow` instead of `findUnique` and `findFirst`, depending on which calls you would like to throw.
+
+If instead you use a custom error handler with the `rejectOnNotFound` property, like these examples:
+
+```js
+// Example 3
+const prisma = new PrismaClient({
+  rejectOnNotFound: (err) => new Error('something'),
+})
+
+// Example 4
+const prisma = new PrismaClient({
+  rejectOnNotFound: {
+    findUnique: (err) => new Error('something'),
+  },
+})
+
+// Example 5
+const prisma = new PrismaClient({
+  rejectOnNotFound: {
+    findFirst: {
+      User: (err) => new Error('User error'),
+      Post: (err) => new Error('Post error'),
+    },
+    findUnique: {
+      User: (err) => new Error('User error'),
+      Post: (err) => new Error('Post error'),
+    },
+  },
+})
+```
+
+You will need to update your method usage to `...OrThrow` and then use a [client extension](/concepts/components/prisma-client/client-extensions) in order to get the same behavior. As an example, the following extension would give the same behavior in Prisma 5 that `Example 5` gave in Prisma 4 and lower.
+
+```js
+import { PrismaClient } from '@prisma/client';
+
+const customErrorFunc = async (model, query, args) => {
+  try {
+    await query(args)
+  } catch (error: any) {
+    if (error.code === 'P2025') {
+      throw new Error(`${model} error`)
+    }
+    throw error;
+  }
+}
+
+const prisma = (new PrismaClient()).$extends({
+  query: {
+    user: {
+      async findFirstOrThrow({ model, query, args }) {
+        return await customErrorFunc(model, query, args)
+      },
+      async findUniqueOrThrow({ model, query, args }) {
+        return await customErrorFunc(model, query, args)
+      },
+    },
+    post: {
+      async findFirstOrThrow({ model, query, args }) {
+        return await customErrorFunc(model, query, args)
+      },
+      async findUniqueOrThrow({ model, query, args }) {
+        return await customErrorFunc(model, query, args)
+      },
+    },
+  },
+})
+```
+
+### <inlinecode>jsonProtocol</inlinecode> out of Preview!
+
+The `jsonProtocol` preview feature is now Generally Available. When upgrading to Prisma 5, make sure to remove `jsonProtocol` from your preview features, if added.
+
+Prisma 4 and lower:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["jsonProtocol"]
+}
+```
+
+Prisma 5:
+
+```prisma
+generator client {
+  provider = "prisma-client-js"
+}
+```
+
+#### Improved error messages
```

Comment:
Again, i would not call them improved, they are mostly just different.

Reply:
Other improvement from @SevInf see Slack https://prisma-company.slack.com/archives/C03MRGP7WLB/p1689065182410019?thread_ts=1689036527.158899&cid=C03MRGP7WLB

Reply:
Added. Thank you both!